client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -663,15 +644,10 @@ public void onFailure(Exception exception) {
          */
         Response get() throws IOException {
             try {
-                //providing timeout is just a safety measure to prevent everlasting waits
-                //the different client timeouts should already do their jobs
-                if (latch.await(timeout, TimeUnit.MILLISECONDS) == false) {
-                    throw new IOException("listener timeout after waiting for [" + timeout + "] ms");
-                }
+                latch.await();
```

Comment:
So we will just wait for as long as it takes for the other side to respond of for the async http client to time out for us?

Reply:
yes